params.go

```diff
@@ -117,3 +156,47 @@ func (b *paramBuilder) addInt32(name *C.char, value int32) {
 		b.err = newOpenSSLError("OSSL_PARAM_BLD_push_int32(" + C.GoString(name) + ")")
 	}
 }
+
+// addBN adds an GO_BIGNUM_PTR to the builder.
+func (b *paramBuilder) addBN(name *C.char, value C.GO_BIGNUM_PTR) {
+	if !b.check() {
+		return
+	}
+	if C.go_openssl_OSSL_PARAM_BLD_push_BN(b.bld, name, value) != 1 {
+		b.err = newOpenSSLError("OSSL_PARAM_BLD_push_BN(" + C.GoString(name) + ")")
+	}
+}
+
+// addBin adds a byte slice to the builder.
+// The slice is converted to a BIGNUM using BN_bin2bn and freed when the builder is finalized.
+// If private is true, the BIGNUM will be cleared with BN_clear_free,
+// otherwise it will be freed with BN_free.
+func (b *paramBuilder) addBin(name *C.char, value []byte, private bool) {
+	if !b.check() {
+		return
+	}
+	bn := C.go_openssl_BN_bin2bn(base(value), C.int(len(value)), nil)
+	if bn == nil {
+		b.err = newOpenSSLError("BN_bin2bn")
+		return
+	}
+	b.bnToFree = append(b.bnToFree, bnParam{bn, private})
+	b.addBN(name, bn)
+}
+
+// addBigInt adds a BigInt to the builder.
+// The BigInt is converted to a BIGNUM using bigToBN and freed when the builder is finalized.
```

Comment:
Slightly more directly saying the BIGNUM is freed rather than the BigInt:

```suggestion
// The BigInt is converted using bigToBN to a BIGNUM that is freed when the builder is finalized.
```